common/lib/dependabot/workspace/git.rb

```diff
@@ -162,6 +162,10 @@ def clean
       sig { params(args: String, kwargs: T.any(T::Boolean, String)).returns(String) }
       def run_shell_command(*args, **kwargs)
         Dir.chdir(path) { T.unsafe(SharedHelpers).run_shell_command(*args, **kwargs) }
+      rescue Dependabot::SharedHelpers::HelperSubprocessFailed => e
+        raise Dependabot::OutOfDisk, e.message if e.message.end_with?("No space left on device")
+        raise Dependabot::OutOfDisk, e.message if e.message.end_with?("Out of diskspace")
+        raise Dependabot::OutOfMemory, e.message if e.message.end_with?("MemoryError")
```

Comment:
This is a good idea but to me this feels like the wrong layer to add it (sounds like you might already be working on a change to this anyways). We already do something like this to check for out of memory errors in [`SharedHelpers.run_helper_subprocess`](https://github.com/dependabot/dependabot-core/blob/dfe50fe71cab4523af0aa1945e483470f639ca15/common/lib/dependabot/shared_helpers.rb#L183), so it seems like maybe this should live in [`SharedHelpers.run_shell_command`](https://github.com/dependabot/dependabot-core/blob/dfe50fe71cab4523af0aa1945e483470f639ca15/common/lib/dependabot/shared_helpers.rb#L401).

Reply:
I think what you saying is correct. Let me change it. Thanks.